calculator.rb

```diff
@@ -0,0 +1,174 @@
+# Calculator
+# Platinum level
+# By Alice Rhomieux
+
+puts "Hello, welcome to the calculator."
+
+# List of supported operations, structured as a hash.
+# Each array is a family of terms that will be accepted for each operation.
+# The operator symbol is always the last element of the class so it can be selected with index [-1]
+# Oops, this is a global variable
+$supported_operations = {
+	addition: ["addition", "add", "plus", "+"],
+	subtraction: ["subtraction", "subtract", "minus", "sub", "-"],
+	multiplication: ["multiplication", "multiply", "times", "*"],
+	division: ["division", "divide", "div", "by", "/"],
+	exponent: ["exponent", "exp", "power of", "power", "**", "^"],
+	modulus: ["modulus", "modulo", "mod", "remainder", "%"]
+}
+
+# Check a number inputted by the user as a string and see if it has a decimal
+# If it has a decimal, convert it to a float, otherwise, convert it to an integer
+def check_num_format(input)
+	if input.include?(".")
+		data = input.delete(",").to_f
+	else
+ 		data = input.delete(",").to_i
+	end
+
+	return data
+end
+
+# Performe a calculation using an operation and two numbers
+# Return the output and the string form of the operator
+def calculate(operation, first_num, second_num)	
+
+	first_number_data = check_num_format(first_num)
+	second_number_data = check_num_format(second_num)
+	
+	if $supported_operations[:addition].include?(operation)
```

Comment:
Good use of a hash to store all operation types within a single variable, and also using `.include?` to avoid having long chains of conditionals. I'm a fan :tada: